snap/snapdir/snapdir.go

```diff
@@ -60,6 +61,89 @@ func (s *SnapDir) ReadFile(file string) (content []byte, err error) {
 	return ioutil.ReadFile(filepath.Join(s.path, file))
 }
 
+func (s *SnapDir) Walk(relative string, walkFn filepath.WalkFunc) error {
+	relative = filepath.Clean(relative)
+	if relative == "" || relative == "/" {
+		relative = "."
+	} else if relative[0] == '/' {
+		// I said relative, darn it :-)
+		relative = relative[1:]
+	}
+	root := filepath.Join(s.path, relative)
+	f, err := os.Open(root)
```

Comment:
Maybe a comment here why it is not `return os.Walk(root, walkFn) at this point?

Reply:
there is no `os.Walk` (that's python :-) ), but there is a `filepath.Walk`; I'll make a comment.